frontend/public/components/overview/index.tsx

```diff
@@ -749,6 +753,27 @@ class OverviewMainContent_ extends React.Component<OverviewMainContentProps, Ove
     });
   }
 
+  createVirtualMachineItems(): OverviewItem[] {
+    const {virtualMachines} = this.props;
+    return _.map(virtualMachines.data, vm => {
+      const buildConfigs = this.getBuildConfigsForResource(vm);
+      const services = this.getServicesForResource(vm);
+      const routes = this.getRoutesForServices(services);
+      const pods = this.getPodsForResource(vm);
```

Comment:
I am not too sure. For now, I simply wanted to comply with the `OverviewItem` contract.

@jelkosz @mareklibra can you please share your thoughts here?

Also note that I've changed

```diff
@@ -1008,7 +1040,7 @@ export type OverviewItem = {
   obj: K8sResourceKind;
   pods?: K8sResourceKind[];
   previous?: PodControllerOverviewItem;
-  readiness: OverviewItemReadiness;
+  readiness?: OverviewItemReadiness;
   routes: K8sResourceKind[];
   services: K8sResourceKind[];
 };
```

in order to avoid rendering "X of Y pods" label on the right side of object list row.

Reply:
In general, there is an initiative to make the VMs to be completely native. It means they should play well with all the features of kubernetes/openshift such as deployments, replica sets etc. We are not there yet but we are getting there.

For this particular objects: 
buildconfig - there has been no effort yet to make this work nor to test it. So I would go with "it does not work" for this case
services: yes, plays well with VMs
routes: yes, plays well with VMs
pods: if the VM is running, there is a pod associated with it in which the qemu process is actually living. So yes.

Reply:
> buildconfig - there has been no effort yet to make this work nor to test it. So I would go with "it does not work" for this case

@jelkosz I can set `buildConfigs` to an empty array (i.e. remove the call to `getBuildConfigsForResource` function).

Still, `OverviewItem.buildConfigs` is currently _not optional_, so we'll end up with _"No Build Configs found for this resource."_ item in the overview details pane anyway:

![vm-overview-list](https://user-images.githubusercontent.com/648971/49604031-1d2d1980-f98d-11e8-8de9-0b5092b267ce.png)

Reply:
@vojtechszocs how big change it would be to enrich the infra to support conditionally not to render the particular resources?